### PR TITLE
[RW-885] One-click unsubscribe links for notifications

### PR DESCRIPTION
**Pull request by orakili**
Refs: RW-885

This replaces the current unsubscribe links with one-click unsubscribe links.

It also adds the proper List-Unsubscribe* headers and removes cookie headers for the POST requests.

## Tests

**Notes** 

- Make sure you have the local stack with mailpit or another tool the RW drupal site can send emails to.
- Make sure you have an instance of the RW API and that the RW local site can index content in it.

**Steps**

1. Checkout the branch, clear the cache, import the config
2. Truncate the subscriptions table: `drush sql-query "TRUNCATE reliefweb_subscriptions_subscriptions"`, `drush sql-query "TRUNCATE reliefweb_subscriptions_queue"`, `drush sql-query "TRUNCATE reliefweb_subscriptions_logs"`
3. Log in, go to your subscriptions page `/user/notifications`, subscribe to the headlines.
4. Create a new headline report, save
5. Run `drush reliefweb_subscriptions:queue -v headlines` then `drush reliefweb_subscriptions:send -d -v`
6. Open the RW mailpit instance, check the email, there should be `Unsubscribe` at the end of the `From` line at the top when opening the email, check there is no red icon (which means it's working). Click on it to reveal the unsubscribe info.
7. Copy the unsubscribe url (without the surrounding `<>`).
8. Run `curl -XPOST "unsubcribe-url-from-above"`
9. Check your subscriptions tab and confirm you are not subscribed to the headlines anymore

### CODE REVIEW

File: html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeController.php [hunk 1]
```diff
@@ -74,6 +145,69 @@ public function unsubscribe(AccountInterface $user) {
...
+      ]);
+
+      $message = $this->t('To add new subscriptions or manage your list, please <a href=":login_link">log in</a>.', [
+        ':login_link' => $login_link->toString(),
+      ]);
+    }
+
+    return [
+      '#type' => 'inline_template',
+      '#template' => '<p><strong>You have successfully unsubscribed.</strong></p><p>You will not longer receive <em>{{ subscription_label }}</em> notifications.</p><p>{{ message }}</p>',
```
Comment (lazysoundsystem) [inline html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeController.php:203]: Typo: "You will no longer receive" (without the 't')

**Review by lazysoundsystem — APPROVED** [review]
Took a while to verify this but in the end I could do it with a disaster subscription and it all works as described.

File: html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeController.php (hunk 1 above)
Comment (orakili) [inline html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeController.php:203]:
```suggestion
      '#template' => '<p><strong>You have successfully unsubscribed.</strong></p><p>You will no longer receive <em>{{ subscription_label }}</em> notifications.</p><p>{{ message }}</p>',
```

Comment (orakili) [reply html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeController.php]: thanks.

**Review by lazysoundsystem — APPROVED** [review]

**Pull request merged**